Algorithms/Medium/142_LinkedListCycleII/Solution.cpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * Definition for singly-linked list.
+ * struct ListNode {
+ *     int val;
+ *     ListNode *next;
+ *     ListNode(int x) : val(x), next(NULL) {}
+ * };
+ */
+class Solution
+{
+public:
+    ListNode *detectCycle(ListNode *head)
+    {
+        if (!head)
+        {
+            printf("no cycle\n");
+            return NULL;
+        }
+        ListNode *tortoise = head, *hare = head;
+
+        while (1)
+        {
+            if (!hare->next || !(hare->next)->next)
+                break; // no cycle
+            // move hare pointer 2 times and tortoise pointer 1 times
+            tortoise = tortoise->next;
+            hare = (hare->next)->next;
+            if (hare == tortoise)
+            {
+                ListNode *T1 = head, *T2 = tortoise;
```

Comment:
Give the variables **T1** and **T2** more meaningful names. Also, the variable naming convention should be consistent. So, stick to camelCase.